core/src/main/scala/com/gu/tableversions/core/model.scala

```diff
@@ -1,8 +1,73 @@
 package com.gu.tableversions.core
 
+import java.net.URI
+
+/**
+  * A Partition represents a concrete partition of a table, i.e. a partition column with a specific value.
+  */
+final case class Partition(columnValues: List[Partition.ColumnValue]) {
+
+  /** Given a base path for the table, return the path to the partition. */
+  def resolvePath(tableLocation: URI): URI = {
+    def normalised(dir: URI): URI = if (dir.toString.endsWith("/")) dir else new URI(dir.toString + "/")
+    if (this == Partition.snapshotPartition) {
+      tableLocation
+    } else {
+      val partitionsSuffix =
+        columnValues.map(columnValue => s"${columnValue.column.name}=${columnValue.value}").mkString("", "/", "/")
+      normalised(tableLocation).resolve(partitionsSuffix)
+    }
+  }
+}
+
+object Partition {
+
+  /** Convenience constructor for single column partitions. */
+  def apply(columnValue: ColumnValue): Partition = Partition(List(columnValue))
```

Comment:
Any reason not to make this:

```scala
def apply(columnValues: ColumnValue*): Partition = Partition(columnValues.toList)
```

Reply:
I'm guessing it doesn't make sense to call that with an empty parameter list?

Reply:
Ah yes, good point!

Reply:
Don't we do that here? https://github.com/guardian/table-versions/pull/2/files/1f7ee1f4992013e443e29c76ae31d44a0a5d68c6#diff-233f9d1b6e81ff49c1ced307784690ccR36

Reply:
That's a good point too! 

So I tried this, but if I make it varargs, it conflicts with the other `apply` method. I'll leave it as it is for now, expecting to review the model API later. (I find it a bit verbose as it is, it would be nice if we could fix that as well)